CONTRIBUTING.rst

```diff
@@ -0,0 +1,500 @@
+   with your new change.
+
+
+GitHub Tips and Tricks
+**********************
+
+Rebasing commits to fix code review issues
+==========================================
+
+You can interactively rebase offending commit(s) to fix code review
+issues.  In your development repo::
+
+  $ git fetch --all
+  $ git rebase --ignore-whitespace origin master
+  $ git rebase -i <offending-commit-id>^
+
+.. note::
+
+   The ``--ignore-whitespace``option stops ``git apply`` (which is
+   called by rebase) from changing any whitespace.
+
+Replace ``pick`` with ``edit`` or remove the line to delete a commit.
+Fix the issue in the code review::
+
+  $ git add [file(s)]
+  $ git rebase --continue
+
+  <update commit comment if needed>
+
+  $ git push --force origin fix_comment_typo
+
+.. _Commit Guidelines:
+
+Commit Guidelines
+*****************
+
+Changes are submitted as Git commits. Each commit message must contain:
+
+* A short and descriptive subject line that is less than 72 characters,
+  followed by a blank line. The subject line must include a prefix that
+  identifies the subsystem being changed, folowed by a colon, and a short
+  title, for example:  ``doc: update wiki references to new site``
+* A change description with your logic or reasoning for the changes, followed
+  by a blank line.
+* A Signed-off-by line, ``Signed-off-by: <name> <email>`` typically added
+  automatically by using ``git commit -s``
+* If the change address a Jira issue, include a line of the form::
+
+      Jira: ZEP-xxx
+
+
+All changes and topics sent to GitHub must be well-formed, as described above.
+
+Comment Message Body
+====================
+
+When editing the commit message, please briefly explain what your change
+does and why it's needed. A change summary of ``"Fixes stuff"`` will be rejected. An
+empty change summary is only acceptable for trivial changes fully described by
+the commit title (e.g., ``doc: fix misspellings in CONTRIBUTING doc``)
+
+The description body of the commit message must include:
+
+* **what** the change does,
+* **why** you chose that approach,
+* **what** assumptions were made, and
+* **how** you know it works -- for example, which tests you ran.
+
+For examples of accepted commit messages, you can refer to the Zephyr GitHub
+`changelog <https://github.com/zephyrproject-rtos/zephyr/commits/master>`__.
+
+Other Commit Expectations
+=========================
+
+* Commits must build cleanly when applied in top of each other, thus avoiding
+  breaking bisectability.
+
+* Commits must pass the *scripts/checkpatch.pl* requirements.
+
+* Each commit must address a single identifiable issue and must be
+  logically self-contained. Unrelated changes should be submitted as
+  separate commits and pull-requests.
+
+* When a change is included in the set to enable other changes, but it will
+  not be part of the final set, let the reviewers know this.
+
+* You can pull-request RFCs (requests for comments) to send work proposals,
```

Comment:
Perhaps:

You can add the "RFC" (request for comments) prefix to your pull request title to send work proposals,